### PR TITLE
MRG, BUG: Fix combine evokeds

### LINKED ISSUE #7865: with weights = 'equal',  mne.combine_evoked computes sum, not mean
Following up on a strange variation in the scale of my evokeds, I realized that
mne.combine_evoked computes the sum, rather than the mean when weights = 'equal' 
(mne version: '0.21.dev0'):

evoked.py, line 876:
  ```
naves = np.array([evk.nave for evk in all_evoked], float)
    if isinstance(weights, str):
        _check_option('weights', weights, ['nave', 'equal'])
        if weights == 'nave':
            weights = naves / naves.sum()
        else:
            weights = np.ones_like(naves)
    else:
        weights = np.array(weights, float)

```
Shouldn't it be 
`weights = np.ones_like(naves) / len(all_evoked) `
instead of 
`weights = np.ones_like(naves) ` ?

The documentation says 
> The weights to apply to the data of each evoked instance. Can also be 'nave' to weight according to evoked.nave, or "equal" to use equal weighting (each weighted as 1/N).

### LINKED ISSUE #7874: Bug in EEG processing and Event Related Potentials Tutorial
#### Describe the bug
I believe the plot_eeg_erp.py tutorial does not combine trials properly. Lines 145-148 should use:
aud_l = epochs["auditory/left"].average()
instead of
aud_l = epochs["auditory","left"].average()

### PR DESCRIPTION
**Pull request by drammock**
closes #7865
closes #7874

still WIP because I need to check the examples/tutorials to make sure they're all consistent with the docstring. Also needs what's new.

### CODE REVIEW

File: tutorials/evoked/plot_eeg_erp.py [hunk 1]
```diff
@@ -125,21 +125,23 @@
 
 ###############################################################################
 # Next, we create averages of stimulation-left vs stimulation-right trials.
-# We can use basic arithmetic to, for example, construct and plot
-# difference ERPs.
+# We can use negative weights in `mne.combine_evoked` to construct difference
+# ERPs.
```
Comment (agramfort) [inline tutorials/evoked/plot_eeg_erp.py:129]:
```suggestion
# of ERPs.
```

File: mne/evoked.py [hunk 2]
```diff
@@ -870,16 +873,15 @@ def combine_evoked(all_evoked, weights):
         if weights == 'nave':
             weights = naves / naves.sum()
         else:
-            weights = np.ones_like(naves)
+            weights = np.ones_like(naves) / len(naves)
     else:
-        weights = np.array(weights, float)
+        weights = np.squeeze(weights).astype(float)
```
Comment (agramfort) [inline mne/evoked.py:878]:
```suggestion
        weights = np.array(weights).astype(float)
```

I hate squeeze magic

File: tutorials/evoked/plot_eeg_erp.py (hunk 1 above)
Comment (hoechenberger) [reply tutorials/evoked/plot_eeg_erp.py:129]: I would say that "difference ERPs" is a valid term on its own, so no need to say "difference **of** ERPs" :)

**Comment by agramfort** [comment]
fair enough. I am not enough an ERP person ;)

>

File: mne/evoked.py (hunk 2 above)
Comment (hoechenberger) [reply mne/evoked.py:878]: I don't see why we would need to use `squeeze` here anyway…?

Comment (drammock) [reply mne/evoked.py:878]: It's not really necessary. My thinking was that we are testing that ndim == 1, and I don't think the docstring explicitly says that weights must be a **1D** array, so I thought using squeeze was being generous to the user. I don't mind changing it back though

**Comment by drammock** [comment]
I've examined all of the examples/tutorials that use `combine_evoked` and it appears that all the figures match up exactly between master and this PR (other than the `plot_eeg_erp` tutorial which I've just corrected).  So should be ready for merge when CIs are happy.

**Comment by agramfort** [comment]
Thx @drammock

**Comment by agramfort** [comment]
@drammock shall we backport this? at least the fix in tutorial?

**Comment by drammock** [comment]
I've been wondering about backporting. The behavior that @SophieHerbst reported was a bug, which is a good reason to backport. On the other hand, this PR also changes the behavior and guidance around how to do evoked subtraction. I'd rather introduce a change like that with a regular release, when people are more likely to read the changelog. I think the latter consideration is stronger for me. @larsoner what is your opinion?

**Comment by hoechenberger** [comment]
@drammock 
I ran into something in our docs that is contradictory and therefore confusing:

> Keeping track of nave is important for inverse imaging, because it is used to scale the noise covariance estimate (which in turn affects the magnitude of estimated source activity). See The minimum-norm current estimates for more information (especially the Whitening and scaling section). For this reason, **combining Evoked objects with either weights='equal' or by providing custom numeric weights should usually not be done if you intend to perform inverse imaging on the resulting Evoked object.**

(from https://mne.tools/dev/auto_tutorials/evoked/plot_10_evoked_overview.html#combining-evoked-objects, highlight by me)

vs

```python
contrast = combine_evoked(evoked, weights=[-1, 1])  # Faces - scrambled
...
# Compute inverse solution on contrast
stc = apply_inverse(contrast, inverse_operator, lambda2, method, pick_ori=None)
```

(from https://mne.tools/dev/auto_examples/datasets/spm_faces_dataset.html#from-raw-data-to-dspm-on-spm-faces-dataset)

As a user, what shall I do? 😭😭😭

**Comment by larsoner** [comment]
> I think the latter consideration is stronger for me. @larsoner what is your opinion?

I'm okay with not backporting

> Keeping track of nave is important for inverse imaging, because it is used to scale the noise covariance estimate (which in turn affects the magnitude of estimated source activity). See The minimum-norm current estimates for more information (especially the Whitening and scaling section). For this reason, combining Evoked objects with either weights='equal' or by providing custom numeric weights should usually not be done if you intend to perform inverse imaging on the resulting Evoked object.

I'm pretty sure `nave` only has an effect for dSPM and sLORETA, and our `weights` math should be correct so actually we should still get this scale factor correct. So maybe we should get rid of this, unless @agramfort sees places it might be problematic that I'm missing.

**Comment by agramfort** [comment]
I agree with @larsoner. I am also not so sure it's problematic to use equal
or custom weights as
the nave of the output should still be correct. But maybe it requires
empirical evidence.

**Comment by drammock** [comment]
actually I guess you're right... it's only `grand_average` where that warning is relevant (it sets `nave=len(evokeds)`). I'll change the tutorial text today.

**Comment by drammock** [comment]
crossref #7881

**Pull request merged**